databroker/src/main.rs

```diff
@@ -321,152 +331,162 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     let args = parser.get_matches();
 
-    // install global collector configured based on RUST_LOG env var.
-    databroker::init_logging();
-
-    info!("Starting Kuksa Databroker {}", version);
-
-    let ip_addr = args.get_one::<String>("address").unwrap().parse()?;
-    let port = args
-        .get_one::<u16>("port")
-        .expect("port should be a number");
-    let addr = std::net::SocketAddr::new(ip_addr, *port);
-
-    let broker = broker::DataBroker::new(version);
-    let database = broker.authorized_access(&permissions::ALLOW_ALL);
-
-    add_kuksa_attribute(
-        &database,
-        "Kuksa.Databroker.GitVersion".to_owned(),
-        option_env!("VERGEN_GIT_SEMVER_LIGHTWEIGHT")
-            .unwrap_or("N/A")
-            .to_owned(),
-        "Databroker version as reported by GIT".to_owned(),
-    )
-    .await;
-
-    add_kuksa_attribute(
-        &database,
-        "Kuksa.Databroker.CargoVersion".to_owned(),
-        option_env!("CARGO_PKG_VERSION").unwrap_or("N/A").to_owned(),
-        "Databroker version as reported by GIT".to_owned(),
-    )
-    .await;
-
-    add_kuksa_attribute(
-        &database,
-        "Kuksa.Databroker.CommitSha".to_owned(),
-        option_env!("VERGEN_GIT_SHA").unwrap_or("N/A").to_owned(),
-        "Commit SHA of current version".to_owned(),
-    )
-    .await;
-
-    if let Some(metadata_filenames) = args.get_many::<String>("vss-file") {
-        for filename in metadata_filenames {
-            read_metadata_file(&database, filename).await?;
-        }
-    }
+    let cores = available_parallelism().unwrap().get();
+    let worker_threads: &usize = args.get_one::<usize>("worker-threads").unwrap_or(&cores);
```

Comment:
What if the number of threads provided is higher than the number of threads allowed?

Reply:
Whats meant by number of threads allowed? You can spawn more threads but if you do not have more cores than it will block the thread. We could limit it to the cores available but I do not think it is necessary.

Reply:
I mean like for example, the user has no idea how many cores the test device has and provides a `woker-threads` parameter higher than the available number of cores.

Reply:
As I understand it is nothing that prevents you from having more threads than cores, but you do not gain anything from it, but most likely you will get a small performance penalty if specifying more than number of cores. I think it makes sense to write some advice in documentation on when it makes sense to have something else than default.

Is a possible use-case that you want to run Databroker with high priority (for minimal latency), but at the same time you do not want it to steal more than a certain share of the capacity and by that reason you state that it only might use X cores.

Reply:
Would there be a value to print something like "Using X cores, Y cores available", either on debug or info level? Info-level if we consider it as relevant info to get when someone has performance problems and we do not want to get a huge debug log.

Reply:
Added info message how many threads are spawned and how many cores are available on the system.